Rubberduck.CodeAnalysis/Inspections/Concrete/EnumerationDeclaredWithinWorksheetInspection.cs

```diff
@@ -0,0 +1,35 @@
+﻿using Rubberduck.CodeAnalysis.Inspections.Abstract;
+using Rubberduck.Parsing.Symbols;
+using Rubberduck.Parsing.VBA;
+using Rubberduck.Parsing.VBA.DeclarationCaching;
+using Rubberduck.Resources.Inspections;
+using Rubberduck.VBEditor.ComManagement;
+using Rubberduck.VBEditor.SafeComWrappers;
+
+namespace Rubberduck.CodeAnalysis.Inspections.Concrete
+{
+    internal class PublicEnumerationDeclaredWithinWorksheetInspection : DeclarationInspectionBase
+    {
+        private readonly IProjectsProvider _projectsProvider;
+
+        public PublicEnumerationDeclaredWithinWorksheetInspection(IDeclarationFinderProvider declarationFinderProvider, IProjectsProvider projectsProvider)
+            : base(declarationFinderProvider)
+        {
+            _projectsProvider = projectsProvider;
+        }
+
+        protected override bool IsResultDeclaration(Declaration declaration, DeclarationFinder finder)
+        {
+            return declaration.Accessibility == Accessibility.Public
+                && declaration.DeclarationType == DeclarationType.Enumeration
```

Comment:
I'm thinking the check needs to be `declaration.Accessibility != Accessibility.Private` in order to flag both implicit and explicit public `Enumerations`

Reply:
:+1: